hack/bicep-types-radius/src/autorest.bicep/src/type-generator.ts

```diff
@@ -358,10 +358,10 @@ export function generateTypes(host: AutorestExtensionHost, definition: ProviderD
     return ObjectTypePropertyFlags.None;
   }
 
-  function parsePropertyFlags(putProperty: Property | undefined, getProperty: Property | undefined) {
+  function parsePropertyFlags(putProperty: Property | undefined, getProperty: Property | undefined, propertyName: string) {
     let flags = ObjectTypePropertyFlags.None;
 
-    if (putProperty && putProperty.required) {
+    if (putProperty && putProperty.required && propertyName !== 'location') {
```

Comment:
We could also check that the propertyType is StringType but I'm not sure how likely we'd be to run into a situation where a 'location' field isn't a string

Reply:
Would this apply to all fields name `location`? Is there a good way to apply this to only fields at the top level?

eg:

- `$.location` is updated to be optional
- `$.properties.foo[0].location` can be required

Reply:
Updated to only apply this check if we're parsing a location field on the resource body. anything nested should still be marked as required